Sources/SpeziViews/Views/ManagedNavigationStack/ManagedNavigationStack.swift

```diff
@@ -0,0 +1,193 @@
+/// In most scenarios, the default identifier is sufficient, but there are some edge cases where a custom identifier needs to be specified,
+/// namely when the ``ManagedNavigationStack`` contains multiple steps with the same `View` type, and you wish to programmatically
+/// navigate to one of them, be it as the stack's starting step (via the `startAtStep` parameter in ``ManagedNavigationStack/init(didComplete:path:startAtStep:_:)``,
+/// or via e.g. ``ManagedNavigationStack/Path/navigateToNextStep(matching:includeIntermediateSteps:)``.
+///
+///
+/// ```swift
+/// struct Onboarding: View {
+///     @AppStorage(StorageKeys.onboardingFlowComplete)
+///     var completedOnboardingFlow = false
+///
+///     var body: some View {
+///         ManagedNavigationStack(didComplete: $completedOnboardingFlow) {
+///             MyOwnView()
+///                 .navigationStepIdentifier("step1")
+///             MyOwnView()
+///                 .navigationStepIdentifier("step2")
+///             // Other views as needed
+///         }
+///     }
+/// }
+/// ```
+///
+/// - Note: When the ``SwiftUICore/View/navigationStepIdentifier(_:)`` modifier is applied multiple times to the same view, the outermost identifier takes precedence.
+///
+/// ## Topics
+/// ### Creating a Managed Navigation Stack
+/// - ``init(didComplete:path:startAtStep:_:)``
+/// - ``NavigationFlowBuilder``
+/// ### Navigation
+/// - ``Path``
+/// ### SwiftUI Environment Values
+/// - ``SwiftUICore/EnvironmentValues/isInManagedNavigationStack``
+/// - ``SwiftUICore/EnvironmentValues/isFirstInManagedNavigationStack``
+public struct ManagedNavigationStack: View {
+    static var logger: Logger { Logger(subsystem: "edu.stanford.spezi.SpeziViews", category: "ManagedNavigationStack") }
+    
+    private let navigationFlow: _NavigationFlow
+    private let isComplete: Binding<Bool>?
+    private let startStepSelector: Path.StepSelector?
+    private var externalPath: Path?
+    @State private var internalPath = Path()
+    
+    /// The effective ``ManagedNavigationStack/Path``
+    var path: Path {
+        externalPath ?? internalPath
+    }
+    
+    @_documentation(visibility: internal)
+    public var body: some View {
+        @Bindable var path = path
+        NavigationStack(path: $path.path) {
+            path.firstStep
+                .environment(\.isFirstInManagedNavigationStack, true)
+                .navigationDestination(for: NavigationStepIdentifier.self) { step in
+                    path.view(for: step)
+                }
+        }
+        .environment(path)
+        .environment(\.isInManagedNavigationStack, true)
+        .onChange(of: ObjectIdentifier(navigationFlow)) {
+            // ensure the model uses the latest views from the initializer
+            path.updateViews(with: navigationFlow.elements)
+        }
+    }
+    
+    /// A `ManagedNavigationStack` is defined by the passed in views defined by the view builder as well as an boolean `Binding`
```

Comment:
maybe use proper ticks and mention the `NavigationFlowBuilder` (or whatever name we'll choose)